src/main/java/com/revature/services/EvalLogicImpl.java

```diff
@@ -16,40 +16,76 @@
 
 	@Autowired
 	private EvalRepository dao;
+	
+
+//CREATE----------------------------
 
+//RETRIEVE--------------------------
 	@Override
 	public Eval getEvalById(int id) {
 		Eval eval = dao.findOne(id);
-		System.out.println(eval);
+		try {
+			System.out.println(eval);
+		} catch (Exception e) {e.printStackTrace();}
 		return eval;
 	}
 	
 	@Override
-	public ArrayList<Eval> getEvalsByBatch(int id){
-		
-		ArrayList<Eval> evals = dao.findAllByBatchId(id);
+	public ArrayList<Eval> getEvalsByBatch(int id, String evalParam){
+		ArrayList<Eval> evals;
+		if (evalParam.equalsIgnoreCase("both")) {
+			evals = dao.findAllByBatchId(id);
+		} else {
+			evals = dao.findAllByBatchIdAndEvalTypeDescription(id, matchEvalType(evalParam));
+		}
 		System.out.println(evals);
 		return evals;
 	}
 	
 	@Override
-	public ArrayList<Eval> getEvalsByWeek(int id, int num){
-		ArrayList<Eval> evals = dao.findAllByWeek(id,num);
+	public ArrayList<Eval> getEvalsByWeek(int id, int num, String evalParam){
+		ArrayList<Eval> evals;
+		if (evalParam.equalsIgnoreCase("both")) {
+			evals = dao.findAllByBatchIdAndWeek(id,num);
+		} else {
+			evals = dao.findAllByBatchIdAndWeekAndEvalTypeDescription(id, num, matchEvalType(evalParam));
+		}
 		System.out.println(evals);
 		return evals;
 	}
-
+	
 	@Override
-	public ArrayList<Eval> getEvalsByPerson(int id) {
-		ArrayList<Eval> evals = dao.findAllByTraineeId(id);
-		System.out.println(evals);
+	public ArrayList<Eval> getEvalsByPerson(int id, String evalParam) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1319)